subprojects/luajit.wrap

```diff
@@ -0,0 +1,5 @@
+[wrap-git]
+directory = luajit
+url = https://github.com/franko/luajit
```

Comment:
personally, I think it would be better if we moved the wrap to a more indepdendent source, like wrapdb.

Reply:
At the moment, wrapdb does not distribute LuaJIT.
If this PR goes forward, feel free to inform me so I can contribute one.

Reply:
Just rebased this branch to latest master with latest dirmonitor changes, if you could do the luajit wrapdb would be nice, even if it is not desirable to add this change to master maybe we can add a master-jit branch that could be kept in sync with master since most of the changes on this branch aren't too invasive. Who knows if in some more years luajit will catch up or a compatibility lib for 5.4 ...

Reply:
good news: the luajit wrap is done

bad news: luajit has a terrible directory structure that assumes in-tree builds and thus prevents it from actually building.

I'll need to talk to some people to figure out how to deal with this

Reply:
I made a LuaJIT repository with a Meson build that is maintained and used by a (few) other people. It works on MacOS, Windows and Linux without known problems.

I sometimes have the impression that some people just want to remove all my personal contributions from this project.

Reply:
> I made a LuaJIT repository with a Meson build that is maintained and used by a (few) other people. It works on MacOS, Windows and Linux without known problems.

Your repo has various modifications from the original LuaJIT source code and the unclean merges have left it in a state where it is hard to reason what was really changed.

Using your meson.build ontop of the official LuaJIT source code is impossible since you changed a few headers (maybe even more) as well as abstracted part of the build process into its own executable, which I doubt would ever be upstreamed.

Its not about not wanting to use your version, it is about using `LuaJIT`, not your flavor of it

> I sometimes have the impression that some people just want to remove all my personal contributions from this project.

If you really want to spin it that way, sure, we've been slowly but surely removing a bunch of code and dependencies added by you as well as moved the build process to depend on a more decentralized method of building.

But that is not what this is about, to me at least.
As previously mentioned, you made changes to LuaJIT which are hard to track and thus it is unknown how far the official version and your version diverge.
If we continue going down that route we will end up with a specialized fork with none of the changes ever upstreamed and upgrading turning into a tedious process of whack-a-diff, moving to using the upstream source code with a patch applied to supply the build system stuff is the better option that will improve the update process as well as allow us to maintain the build system stuff independently of LuaJIT.

If you went missing tomorrow we would need to do this anyway, since you are the only one with control over LuaJIT repo and we really don't need that point of failure.